src/compiler/utilities.ts

```diff
@@ -1966,7 +1966,7 @@ namespace ts {
         let name: Expression | BindingName | undefined;
         let decl: Node | undefined;
         if (isVariableDeclaration(node.parent) && node.parent.initializer === node) {
-            if (!isInJSFile(node) && !isVarConst(node.parent)) {
+            if (!isInJSFile(node) && !(isVarConst(node.parent) && isFunctionSymbol(node.parent.symbol))) {

```

Comment:
`isFunctionSymbol` does some redundant or unneeded work -- you can get the same effect by checking  `isFunctionLikeDeclaration(node.parent.initializer)` instead

Reply:
Nice!